server/src/main/java/org/elasticsearch/action/admin/indices/validate/query/QueryExplanation.java

```diff
@@ -110,4 +151,43 @@ public static QueryExplanation readQueryExplanation(StreamInput in)  throws IOEx
         exp.readFrom(in);
         return exp;
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        if (getIndex() != null) {
+            builder.field(INDEX_FIELD, getIndex());
```

Comment:
I tend to use the members rather than the getters for these but either way is ok by me!

Reply:
I used to prefer that too. Then I saw in some classes that the getters also had some logic which I wanted to use. So I started using these...